site/src/components/CopyButton/CopyButton.tsx

```diff
@@ -23,36 +24,44 @@ export const Language = {
 /**
  * Copy button used inside the CodeBlock component internally
  */
-export const CopyButton: FC<CopyButtonProps> = ({
-  text,
-  ctaCopy,
-  wrapperStyles,
-  buttonStyles,
-  tooltipTitle = Language.tooltipTitle,
-}) => {
-  const { isCopied, copy: copyToClipboard } = useClipboard(text);
+export const CopyButton = forwardRef<HTMLButtonElement, CopyButtonProps>(
+  (props, ref) => {
+    const {
+      text,
+      ctaCopy,
+      wrapperStyles,
+      buttonStyles,
+      onClick: outsideOnClick,
+      tooltipTitle = Language.tooltipTitle,
+    } = props;
+    const { isCopied, copyToClipboard } = useClipboard(text);
 
-  return (
-    <Tooltip title={tooltipTitle} placement="top">
-      <div css={[{ display: "flex" }, wrapperStyles]}>
-        <IconButton
-          css={[styles.button, buttonStyles]}
-          onClick={copyToClipboard}
-          size="small"
-          aria-label={Language.ariaLabel}
-          variant="text"
-        >
-          {isCopied ? (
-            <Check css={styles.copyIcon} />
-          ) : (
-            <FileCopyIcon css={styles.copyIcon} />
-          )}
-          {ctaCopy && <div css={{ marginLeft: 8 }}>{ctaCopy}</div>}
-        </IconButton>
-      </div>
-    </Tooltip>
-  );
-};
+    return (
+      <Tooltip title={tooltipTitle} placement="top">
+        <div css={[{ display: "flex" }, wrapperStyles]}>
+          <IconButton
+            ref={ref}
+            css={[styles.button, buttonStyles]}
+            size="small"
+            aria-label={Language.ariaLabel}
+            variant="text"
+            onClick={(event) => {
+              void copyToClipboard();
+              outsideOnClick?.(event);
```

Comment:
Why do we need this?

Reply:
More just for flexibility. It might be because I've been in a Backstage mindset, where we need a lot more API flexibility, but my thinking was this would let someone attach an `onClick` handler without overriding the main copy functionality

Reply:
In the Coder app context + this component context, what would be a common use case for this?

Reply:
For this context, it's over-engineering. I'll remove it